pkg/shell/base_index.js

```diff
@@ -143,6 +144,42 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
                 }
             }
 
+            function domChangeHandler(mutation) {
```

Comment:
The idea was to keep this fairly light-weight and non-intrusive. Do you have a more concrete reference to code that's like this? If part of the DOM gets removed (i.e. an iframe), we just want to react to that change. At the user level we then don't have to worry about implementation details, we just remove the iframe.